### PR TITLE
black: 21.7b0 -> 21.8b0

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Bumped black and pathspec (due to a new requirement from black)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/black/default.nix [hunk 1]
```diff
@@ -10,6 +10,7 @@
 , mypy-extensions
 , pathspec
 , parameterized
+, platformdirs
```
Comment (sveitser) [inline pkgs/development/python-modules/black/default.nix:13]: Can the `appdirs` dependency be removed? https://github.com/psf/black/pull/2375

**Review by sveitser — CHANGES_REQUESTED** [review]
The rest looks good to me.

File: pkgs/development/python-modules/black/default.nix [hunk 2]
```diff
@@ -20,13 +21,13 @@
...
   pname = "black";
-  version = "21.7b0";
+  version = "21.8b0";
 
   disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "06d27adq6v6p8wspi0wwqz2pnq34p5jhnqvijbin54yyj5j3qdy8";
+    sha256 = "sha256-VwYI0oqjrxeSuYxKM326xjZ4d7R7EriKtCCVz8GmJ8I=";
```
Comment (sveitser) [inline pkgs/development/python-modules/black/default.nix:30]: `hash = ... ` is less redundant and more common

Comment (fabaff) [reply pkgs/development/python-modules/black/default.nix:30]: We should stick to `sha256` for now. As long as there is no decision made about this (https://github.com/NixOS/nixpkgs/pull/112388) and support present in the tools for auto-conversion (e.g., `nix-update`).

Comment (sveitser) [reply pkgs/development/python-modules/black/default.nix:30]: Ok. Thanks for the explanation and for introducing me to `nix-update` :smile:

**Review by sveitser — APPROVED** [review]
Changes look good to me now, thanks.

**Pull request merged**